build.sbt

```
@@ -63,6 +64,47 @@ lazy val equella = (project in file("."))
              UpgradeInstallation,
              learningedge_config)
 
+checkJavaCodeStyle := {
+  import com.etsy.sbt.checkstyle._
+  val rootDirectory       = (baseDirectory in LocalProject("equella")).value
+  val rootTargetDirectory = (target in LocalProject("equella")).value
+  def countErrorNumber: Int = {
+    val outputFile = new File("target/checkstyle-report.xml")
+    if (outputFile.exists()) {
+      val report = scala.xml.XML.loadFile(outputFile)
+      (report \ "file").flatMap { file =>
+        (file \ "error").map { _ =>
+          {
+            1
+          }
+        }
+      }.sum
+    } else {
+      Int.MaxValue
```

This should probably throw an exception (e.g. `FileNotFoundException`). Just returning a really high value is going to be misleading down below.